client/state/themes/actions.js

```diff
@@ -194,39 +195,79 @@ export function receiveThemes( data, site, queryParams, responseTime ) {
 	};
 }
 
-export function activate( theme, site, source = 'unknown' ) {
+/**
+ * Returns an action object to be used in signalling that a theme activation
+ * has been triggered
+ *
+ * @param  {String}  themeId Theme to be activated
+ * @param  {Object}  siteId Site used for activation
+ * @return {Object}  Action object
+ */
+export function themeActivation( themeId, siteId ) {
+	return {
+		type: THEME_ACTIVATE_REQUEST,
+		themeId,
+		siteId,
+	};
+}
+
+/**
+ * Returns an action object to be used in signalling that a theme activation
+ * has been successfull
+ *
+ * @param  {Object}  theme Theme received
+ * @param  {Number}  siteId Site used for activation
+ * @return {Object}  Action object
+ */
+export function themeActivated( theme, siteId ) {
```

Comment:
Okay. I've had a bit more of a look into other parts of the state tree and their nomenclature.

Judging from those other parts, I'm quite convinced we should inline `themeActivated` _and_  `themeActivationFailed` _and_ `themeActivation`. (As I said, I think they're still self-explanatory enough even when inlined, because of the action name constants in their `type` fields.)

This leaves us with `activateTheme` (which is good as-is) and `themeActivationSuccess`, which I think we should rename so it isn't easily confused with an action that only dispatches `THEME_ACTIVATE_REQUEST_SUCCESS`. This one isn't quite easy to name if we want it to be a verb, but I'm leaning towards `themeActivated`, at least for now. Maybe not perfect, but I think it's fine for the start.

Anyway, I think it actually makes it easier for me to reason about these actions with some of them inlined and thus fewer potential (mental) naming collisions.

Hope you're okay with that 😄